spec/support/corpus.rb

```diff
@@ -124,6 +116,40 @@ def checkout
 
     private
 
+      MissingExpectedException = Class.new(StandardError)
+
+      # Count mutations and check error results against whitelist
+      #
+      # @param path [Pathname] path responsible for exception
+      #
+      # @return [Fixnum] mutations generated
+      def count_mutations_and_check_errors(path)
+        relative_path = path.relative_path_from(repo_path)
+
+        count = count_mutations(path)
+
+        expected_errors.assert_success(relative_path)
```

Comment:
It looks like `count_mutations` is a pure function, so does it matter if this assertion is done before or after it? I generally like to put assertions as early as possible in the chain, in order to safe unnecessary work.

Also, it'll allow you to return `count_mutations(path)` instead of creating a synthetic variable to return.